helium: adjust disable date to today

dumbmoron opened the pull request:
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

This PR serves as less of a request, and more of a question. I was wondering if the removal of the `helium` cask could be expedited beyond what is its current expiry in two years.

The only official policy I could find regarding this, is *Deprecating, Disabling and Removing Casks* [1], which states that: 
> The most common reasons for disabling a cask are:
> ...
> it cannot be installed on any supported OS versions
> ...
> Unpopular casks (e.g. have fewer than 300 [analytics installs in the last 90 days](https://formulae.brew.sh/analytics/cask-install/90d/)) can be disabled immediately for any of the reasons above, e.g. the upstream URL has been removed. They can be manually removed three months after their disable date.

At the time of writing, the `helium` cask has 132 (331) downloads in the past 90 days (year). It is deprecated due to being unsigned, and does not have any ARM version. It has not been materially changed since it was added in 2015, and the mobile application it's supposed to interface with[2], has been gone from the Play Store for at least four years[3].

The motivation behind this request is that I am one of the maintainers of Helium[4, 5], a browser that we are planning on releasing soon. We would prefer to, if possible, use *this* name on Homebrew as a method of downloading it, sooner than in two years' time. 

[1] https://docs.brew.sh/Deprecating-Disabling-and-Removing-Casks#disabling
[2] https://play.google.com/store/apps/details?id=com.koushikdutta.backup
[3] https://web.archive.org/web/20211022113428/https://play.google.com/store/apps/details?id=com.koushikdutta.backup
[4] https://helium.computer
[5] https://github.com/imputnet/helium-chromium

p-linnane commented:
Even if we were to move the disable date today (which we will not), this wouldn't be removed from Homebrew/cask for another year after being disabled. The earliest you could try to claim the `helium` cask token would be September 2027. It's probably better to submit your cask as `helium-browser` or something.

p-linnane commented:
> > which we will not
> 
> why not?

Because I don't see a compelling reason to.
> 
> > this wouldn't be removed from Homebrew/cask for another year after being disabled
> 
> why not? i doubt a single human has used this package in last 5-9 years.

Because this is how our project operates.

> does this policy no longer apply? shouldn't the [deprecation info page](https://docs.brew.sh/Deprecating-Disabling-and-Removing-Casks#disabling) be updated, then?

Please notice that it says "can be disabled". That call is made by maintainers.

SMillerDev commented:
I don't think the Homebrew project has ever replaced one piece of software with another under the same token in this way.
It would undermine user trust to switch like that, no matter how (un)popular the original software is.

I think you're better off with `helium-browser` or `helium-chromium` since that's the name your repo uses.

tiagozip commented:
> It would undermine user trust to switch like that, no matter how (un)popular the original software is.

i don't really think this is a matter of it being just "unpopular". as op mentioned,

> It is **deprecated** due to being unsigned, [...]. It has not been materially changed since it was added in **2015**, and **the mobile application it's supposed to interface with[2], has been gone from the Play Store for at least four years**

it has most likely **not been used by any real humans in multiple years** (and will probably never be. guessing the download count is just mirrors or something).

keeping the cask name to just "helium" would make it much easier to memorize and type. i think almost all popular package registries have done this.

dumbmoron commented:
> I don't think the Homebrew project has ever replaced one piece of software with another under the same token in this way.

@SMillerDev You can find many instances of casks being replaced with unrelated software in this repo's git history, for example to name a few that I found: spatial (removed 828c91736357a9f92685766448574cd21326bc91, added 2672ff26fd7991a6e25a3e1e38029fe7f44b22f6), digital (removed 5c308fccfa6529a9d762ad054f6cc3f6d8d64aff, added 6a36c930cbb37a14db92d6f463a9327b490693b7), conductor (removed b6d0d30330dc55ba77d15bf8082104a40fb09283, added 8318d9acd9270abeac8c4913d1ee75ff800ed5e0), bloom (removed cca77b9bf113a7278b85fa7ccd09902bef31737e, added 97b2f474cafb97cb173029d0cc4ea24f1287d2e2), feather (removed 21c4d0ac755e333e15eecfbab1fa080d2f7be5db, added d0f8393a929a1c51e2450ee057bfa32c8f357032).

But the point of this PR was not to instantly replace the cask (as you can see by the diff), but rather to ask whether it would be able to shorten the waiting time of a to-be-deleted-anyways cask that could use it right now. Clearly, the answer is no.

Anyways, you are the maintainer, and I'm not. So do whatever you want. ¯\\_(ツ)\_/¯

p-linnane commented:
@SMillerDev's point was related to @wukko insinuating the package should be removed immediately, rather than on our normal timeline, because no one has used it in years. A cask token can picked up by a new project after the old one is fully removed, as you've shown in multiple instances.

The 👎 brigading from Helium users/contributors is pretty sad, but it doesn't change anything. Homebrew maintainers make decisions for the Homebrew project, not entitled users of yet another Chromium browser.
